### PR TITLE
Update typings for json2csv@4.4.x

### PR DESCRIPTION
**Pull request by Dangoo**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Documentation of AsyncParser](https://github.com/zemirco/json2csv/tree/v4.4.0#json2csv-async-parser-streaming-api)
- [x] Increase the version number in the header if appropriate.
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~

This PR adds types for the AsyncParser method added in json2csv@4.4.0 (sadly not mentioned in change log…).

If you have any comments pls. let me know.

### CODE REVIEW

**Comment by juanjoDiaz** [comment]
I'm currently out but I'll take a look at this on Monday.

File: types/json2csv/index.d.ts
Comment (juanjoDiaz) [inline types/json2csv/index.d.ts]: Although I appreciate the additions, I'm not convinced that the name of the main maintainer should be completely removed in favour of someone not related to the development of the library 🙂

File: types/json2csv/index.d.ts
Comment (juanjoDiaz) [inline types/json2csv/index.d.ts]: `data` can also be a `stream.Readable`.

File: types/json2csv/JSON2CSVAsyncParser.d.ts
Comment (juanjoDiaz) [inline types/json2csv/JSON2CSVAsyncParser.d.ts]: This should be `stream.Readable` instead of `ReadStream`.

File: types/json2csv/JSON2CSVAsyncParser.d.ts
Comment (juanjoDiaz) [inline types/json2csv/JSON2CSVAsyncParser.d.ts]: This should be `stream.Writable` instead of `WriteStream `.

File: types/json2csv/index.d.ts
Comment (juanjoDiaz) [inline types/json2csv/index.d.ts]: `data` can also be a `stream.Readable`.

File: types/json2csv/json2csv-tests.ts
Comment (juanjoDiaz) [inline types/json2csv/json2csv-tests.ts]: Does this even work if there is no file on `'/path/to/input'`?

File: types/json2csv/json2csv-tests.ts
Comment (juanjoDiaz) [inline types/json2csv/json2csv-tests.ts]: There should also be a test where data is an object and where data is a readable stream

File: types/json2csv/json2csv-tests.ts
Comment (juanjoDiaz) [inline types/json2csv/json2csv-tests.ts]: I find this test a bit misleading.

You typically with either uses an output or use the promise, but not both.

**Review by juanjoDiaz — CHANGES_REQUESTED** [review]
Thanks for your contribution.

If left a few comments.

File: types/json2csv/index.d.ts
Comment (Dangoo) [reply types/json2csv/index.d.ts]: Yeah, you're totally right

File: types/json2csv/index.d.ts
Comment (Dangoo) [reply types/json2csv/index.d.ts]: @juanjoDiaz  Just for me to catch up, how does the synchronous parser handle a stream input?

File: types/json2csv/json2csv-tests.ts
Comment (Dangoo) [reply types/json2csv/json2csv-tests.ts]: [Readme](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) says:
> This code does **not** run, but it is type-checked.

So should not be a problem here

File: types/json2csv/json2csv-tests.ts
Comment (Dangoo) [reply types/json2csv/json2csv-tests.ts]: This example is just taken from the packages [README](https://github.com/zemirco/json2csv#json2csv-async-parser-streaming-api).
Thought the examples would make okayish test cases here.

@juanjoDiaz Can you provide a more fitting example?

**Comment by Dangoo** [comment]
@juanjoDiaz Thanks for your review!
Applied the requested changes.
Just left two discussions open, you can simply resolve them or reply :)

BTW: Are you a maintainer of `json2csv`? Could you then maybe add the asyncParser feature to the change log of `v4.4.0`?

File: types/json2csv/index.d.ts
Comment (juanjoDiaz) [reply types/json2csv/index.d.ts]: My bad. Only the `asyncParser` can. 🙂

File: types/json2csv/json2csv-tests.ts
Comment (juanjoDiaz) [reply types/json2csv/json2csv-tests.ts]: Probably I should review our own examples then 😅 

Typically you would do either `asyncParser.fromInput(input).toOutput(output)` or `asyncParser.fromInput(input).promise()`. The promise loads the whole resulting CSV in memory so it kind if defeat the purpose of streaming directly to an output (I'll revise the docs)

**Comment by juanjoDiaz** [comment]
I'm the main maintainer atm but not the one tagging releases.

I'll make sure that it's included in the changelog.

Thanks for the good work!

File: types/json2csv/json2csv-tests.ts
Comment (juanjoDiaz) [reply types/json2csv/json2csv-tests.ts]: Everything looks great so as soon as you split this into 2 examples, I'll approve the PR.

Thanks again!

Comment (Dangoo) [reply types/json2csv/json2csv-tests.ts]: Done.

**Review by juanjoDiaz — APPROVED** [review]
Looks good!

**Pull request merged**
